diff --git a/src/fdcache_entity.h b/src/fdcache_entity.h
index 6416b01..d2776b3 100644
--- a/src/fdcache_entity.h
+++ b/src/fdcache_entity.h
@@ -18,6 +18,14 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
+// Constants are defined in glibc >= 2.18
+#ifndef FALLOC_FL_KEEP_SIZE
+#define FALLOC_FL_KEEP_SIZE    0x1
+#endif
+#ifndef FALLOC_FL_PUNCH_HOLE
+#define FALLOC_FL_PUNCH_HOLE   0x2
+#endif
+
 #ifndef S3FS_FDCACHE_ENTITY_H_
 #define S3FS_FDCACHE_ENTITY_H_
 